odxtools/diaglayer.py

```diff
@@ -261,7 +262,7 @@ def _resolve_references(self, odxlinks: OdxLinkDatabase) -> None:
 
         self._communication_parameters = NamedItemList[CommunicationParameterRef](
             short_name_as_id,
-             list(self._compute_available_commmunication_parameters_by_name().values())
```

Comment:
ok, I see: the dict was actively harmful...

Reply:
> ok, I see: the dict was actively harmful...

I believe, that the same problem applies to the `get_inherited_data_object_properties_by_name` and 'get_inherited_services_by_name`. Shouldn't we update these methods as well?

Reply:
possibly: I'm not 100% about the semantics  specified by the standard, but I consider it likely that DOPs and services specified in derived diag layers override those of their parents. (for comparams  it is probably actually the same, but instead of using a simple name to decide if a comparam ought to be overriden, we need to use `(short_name, protocol, is_functional)`?)

Reply:
I'll keep this in mind in case I observe some strange behavior. Since everything has worked so far with the pdx files I have used, this topic is a bit too hot for me to address this in this PR right away.